src/harness/reference_models/pre_iap_filtering/pre_iap_util.py

```diff
@@ -0,0 +1,186 @@
+      if pal_record['palId'] == ppa_pal_ids[pal_id]:
+        pal_frequency = pal_record['channelAssignment']['primaryAssignment']
+        pal_frequencies.append(pal_frequency)
+        break
+    else:
+      # Raise an exception if no matching PAL ID is found
+      raise Exception('PAL record %s not found for PPA ID %s' %(pal_record['palId'], ppa_record['id']))
+
+  # Compare the frequencies of all the PAL records of the PPA
+  for pal_frequency in pal_frequencies:
+    if pal_frequency != pal_frequencies[0]:
+      # If the frequency of the PAL records are not matching raise an exception
+      raise Exception('The frequency range in all PAL records are not same for PPA ID %s' %ppa_record['id'])
+
+  # Return the first frequency range of the first matching PAL ID
+  return pal_frequencies[0]
+
+def getFssNeighboringCbsdsWithGrants(cbsds, fss_record, distance):
+  """Get the list of all CBSDs that lie within 150 KMs of FSS and have at least one grant.
+
+  Args:
+    cbsds :  List of CbsdData dictionaries as defined in the SAS-SAS specification.
+    fss_record: A FSS record dictionary.
+  Returns:
+    List of CBSDs lying within 150 KMs of FSS and having at least one grant.
+  """
+
+  neighboring_cbsds_with_grants = []
+  for cbsd in cbsds:
+    distance_km, _, _ = vincenty.GeodesicDistanceBearing(
+        fss_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+        fss_record['record']['deploymentParam'][0]['installationParam']['longitude'],
+        cbsd['registrationRequest']['installationParam']['latitude'],
+        cbsd['registrationRequest']['installationParam']['longitude'])
+    # Get the list of cbsds that are within 150kms from the FSS entity
+    if distance_km <= distance and cbsd['grants']:
+        neighboring_cbsds_with_grants.append(cbsd)
+  return neighboring_cbsds_with_grants
+
+
+def getFssNeighboringGwbl(gwbl_records, fss_records): 
+  """Get the list of all fss that lie within 150 KMs of GWBL and operating below 3700 MHz
+
+  Args:
+    fss_records:  List of FSS records dictionary.
+    gwbl_records: List of GWBL record dictionary.
+  Returns:
+    List of FSSs lying within 150 KMs of GWBL and operating below 3700000000.
+  """
+
+  list_of_fss_neighboring_gwbl = []
+  for fss_record in fss_records:
+    for gwbl_record in gwbl_records:
+      # Get the distance of the FSS entity from the GWBL polygon
+      distance_km, _, _ = vincenty.GeodesicDistanceBearing(
+          fss_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+          fss_record['record']['deploymentParam'][0]['installationParam']['longitude'],
+          gwbl_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+          gwbl_record['record']['deploymentParam'][0]['installationParam']['longitude']) 
+      # Get the list of FSS entity that are 150kms from the GWBL area
+      if distance_km <= FSS_GWBL_PROTECTION_DISTANCE and fss_record['record']['deploymentParam'][0]\
```

Comment:
And here you can remove the frequency check since done before.

Reply:
Removed   the frequency check